src/libtiled/tilelayer.cpp

```diff
@@ -312,12 +312,24 @@ void TileLayer::rotate(RotateDirection direction)
     mGrid = newGrid;
 }
 
-void TileLayer::rotateHexagonal(RotateDirection direction)
+void TileLayer::rotateHexagonal(RotateDirection direction, Map::StaggerIndex staggerIndex, Map::StaggerAxis staggerAxis)
 {
-    int newWidth = mHeight;
-    int newHeight = mWidth;
+    Hex botRight(mWidth, mHeight, staggerIndex, staggerAxis);
+    Hex topRight(mWidth, 0, staggerIndex, staggerAxis);
+    Hex center(mWidth / 2, mHeight / 2, staggerIndex, staggerAxis);
+
+    botRight -= center;
+    topRight -= center;
+
+    botRight.rotate(RotateRight);
+    topRight.rotate(RotateRight);
+
+    int newWidth = topRight.toStaggered(staggerIndex, staggerAxis).x() * 2 + 2;
+    int newHeight = botRight.toStaggered(staggerIndex, staggerAxis).y() * 2 + 2;
```

Comment:
It is not immediately obvious to me why this works (like, why the new area is both minimal but also large enough). Could you share a few words about it?